peepingtom/io_/read/read.py

```diff
@@ -79,37 +79,45 @@ def find_files(paths, filters=None, recursive=False, max=None):
     return files[:max]
 
 
-def read_to_datablocks(paths, filters=None, recursive=False, strict=False, max=None, **kwargs):
+def read(paths,
+         filters=None,
+         name_regex=None,
+         mode=None,
+         recursive=False,
+         strict=False,
+         max=None,
+         **kwargs):
     """
-    read generic path(s) into the appropriate datablocks
-    strict: if set to False, ignore failures and read what possible
-    """
-    datablocks = []
-    for file in find_files(paths, filters=filters, recursive=recursive, max=max):
-        try:
-            datablocks.extend(read_file(file, **kwargs))
-        except ParseError:
-            if strict:
-                raise
-    if not datablocks:
-        raise ParseError(f'could not read any data from {paths}')
-    return datablocks
+    read generic path(s) and construct a peeper accordingly
 
-
-def read(paths, mode=None, **kwargs):
-    """
-    read generic path(s) and construct a dataset accordingly
-    mode:
+    filters: a regex string or iterable thereof used to select filenames [default: '.*']
```

Comment:
should be changed to a filter function rather than regex string

Reply:
This is not something that can be passed from command line though. We can make it possible from `pt.read`, but not here!